src/server/game/Entities/Unit/Unit.cpp

```diff
@@ -422,6 +422,7 @@ Unit::~Unit()
     ASSERT(m_removedAuras.empty());
     ASSERT(m_gameObj.empty());
     ASSERT(m_dynObj.empty());
+    ASSERT(!_gameClientMovingMe || _gameClientMovingMe->GetBasePlayer() == this);
```

Comment:
not sure why this assert is even here ?

Reply:
This is here to make sure that client control code properly cleans up after itself

Reply:
do you know where the cleanup call is done ? Is there anything in RemoveFromWorld ?

Reply:
the clean up is in [GameClient::RemoveAllowedMover](https://github.com/TrinityCore/TrinityCore/blob/3.3.5/src/server/game/Server/GameClient.cpp#L37). I'll look into this